dashboard/src/app/dashboards/CreateDashboardDialog.tsx

```diff
@@ -73,14 +79,38 @@ export function CreateDashboardDialog() {
     }
   };
 
-  return (
-    <Dialog open={open} onOpenChange={handleOpenChange}>
-      <DialogTrigger asChild>
-        <Button variant='outline' className='gap-2'>
+  const createButton = (
+    <Button variant='outline' className='gap-2' disabled={!dashboardStats.canCreateMore}>
+      {dashboardStats.canCreateMore ? (
+        <>
           <Plus className='h-4 w-4' />
           Create Dashboard
-        </Button>
-      </DialogTrigger>
+        </>
+      ) : (
+        <>
+          <Lock className='h-4 w-4' />
+          Create Dashboard
+        </>
+      )}
+    </Button>
+  );
+
+  const triggerElement = !dashboardStats.canCreateMore ? (
+    <Tooltip>
+      <TooltipTrigger asChild>
+        <span className='inline-block'>{createButton}</span>
+      </TooltipTrigger>
+      <TooltipContent>
+        <p>You've reached your dashboard limit. Upgrade your plan to create more dashboards.</p>
+      </TooltipContent>
+    </Tooltip>
+  ) : (
+    createButton
+  );
```

Comment:
Nitpick: I prefer happy path first - as I initially skimmed it, I has missed the `!` and got a little confused why it would show that you've reached your limit
```suggestion
  const triggerElement = dashboardStats.canCreateMore ? (
    createButton
  ) : (
    <Tooltip>
      <TooltipTrigger asChild>
        <span className='inline-block'>{createButton}</span>
      </TooltipTrigger>
      <TooltipContent>
        <p>You've reached your dashboard limit. Upgrade your plan to create more dashboards.</p>
      </TooltipContent>
    </Tooltip>
  );
```